src/player.cpp

```diff
@@ -0,0 +1,63 @@
+#include "player.h"
+#include <iostream>
+
+// Helper function
+/**
+ * @brief Trim trailing whitespace from a string
+ * 
+ * @param s string to trim
+ */
+void rtrim(std::string& s) {
+  s.erase(s.find_last_not_of(" \n\r\t") + 1);
+} 
+
+Player::Player(bool is_cpu): is_cpu_(is_cpu) {
+  id_ = 0;
+  score_ = 0;
+  last_move_ = 0;
+  board_size_ = 0;
+}
+
+void Player::init(unsigned int id, unsigned int board_size) {
+  id_ = id;
+  board_size_ = board_size;
+  score_ = 0;
+  last_move_ = 0;
+  moves_.clear();
+  for (int i = 1; i <= board_size; ++i) {
+    moves_.insert(i);
+  }
+}
+
+bool Player::is_valid_move(unsigned int number) {
+  return (moves_.find(number) != moves_.end());
+}
+
+unsigned int Player::move(unsigned int number) {
+  if (!is_valid_move(number)) {
+    return 0;
+  }
+  moves_.erase(number);
+  last_move_ = number;
+  return number;
+}
+
+std::string Player::get_moves_string() {
+  std::vector<bool> moves(board_size_);
```

Comment:
if this is constant size then why don't we use a `std::array` ;)

this is a nit I'm an embedded guy sorry

Reply:
#safetyfirst

Reply:
oh nvm it doesn't work cause board_size can change?

Reply:
Board size can change so it won't let me define a constant size array.